### ISSUE TITLE
Can't switch subtitle from primary to secondary position (or display in both)

### ISSUE THREAD

**Issue by gbezyuk** [iina/iina#4457]
**System and IINA version:**

- macOS Mojave 10.14.6 (18G103)
- IINA 1.3.0 Build 132 mpv 0.34.1

**Steps to reproduce:**
— while a video file with at least one subtitle track(s) is open
— select some subtitle track as _primary_
— try to select **the same** subtitle track as a _second subtitle_

**Expected behaviour:**
either
— the same subtitle displays now in both primary and secondary position
or
— subtitle position switched from primary to secondary position

**Actual behaviour:**
— there is no change, no reaction on the secondary subtitle selection whatsoever: primary subtitle is still active, and secondary subtitle is still empty.

**How often does this happen?**
It's a stable problem, it happens always.

**Comment by low-batt** [comment]
Reproduced for me.

Similar to many open source projects IINA is layered on top of other projects. For audio/video playback and subtitle rendering IINA is using a library from the [mpv](https://mpv.io/) project. For this reason the issue template contains the following checkbox:
> \
> - [ ] MPV does not have this problem.

The reason for this question is that problems involving audio/video playback or subtitle rendering that reproduce using the mpv player need to be reported to that project. Once the mpv project has fixed the problem IINA will need to update the version of the mpv library it is using to pickup the fix.

Back to the issue at hand. Checking the log shows this message from mpv:
```text
23:08:31.719 [mpv][e] [cplayer] error: Track 2 is already selected.
```

Seems mpv does not permit displaying the same subtitle track as both the primary and secondary subtitle track.

I'm assuming this is causing trouble in the case where you want to make the primary subtitle the secondary subtitle and then select another track to be the primary track?

Given mpv's restriction I consider it a bug in IINA in that it is presenting the user an option to select a subtitle track that is already in use. IINA needs to be updated to adhere to mpv's requirement and not provide the user with an option to select an active track.

**Comment by gbezyuk** [comment]
Honestly, I wasn't even quite familiar with the fact of existence of MPV, not to mention of whether it could render secondary subtitles (which is a disappointingly rare feature).

> IINA needs to be updated to adhere to mpv's requirement and not provide the user with an option to select an active track.

I'd disagree. Let the user select the track, just move it from primary to secondary position (or visa versa) in case it was already selected for the different channel. That would produce a much better UX, if you ask me.

**Comment by low-batt** [comment]
Yes, moving the track to the other subtitle position is another option. I'm leaving it to the senior developers to make a call on how to fix this. It is clear the current behavior is wrong.

The mpv player is a media player for the command line. So normally you start it from [Terminal](https://support.apple.com/guide/terminal/welcome/mac), passing options and the file to play on the command line. From the home page:
> While mpv strives for minimalism and provides no real GUI, it has a small controller on top of the video for basic control. 

The project also publishes a library that other projects can use. There are a number of projects that provide [GUI frontends](https://github.com/mpv-player/mpv/wiki/Applications-using-mpv#gui-frontends) for mpv.

This means a lot of the functionality listed in the [mpv manual](https://mpv.io/manual/stable/) is available in IINA. If IINA does not provide a GUI control for a feature you may still be able to use it by adding the mpv option in IINA's settings under the `Advanced` tab.

**Comment by gbezyuk** [comment]
One more related thing: some subtitle tracks force subtitle position at times, which leads to subtitle overlay in case both primary and secondary subtitles are enabled. It would be great to have an option to disable such jumping and fix the subtitles in place, ignoring their internal repositioning instructions.
Also, hotkeys for changing subtitle size would be really nice.

**Comment by low-batt** [comment]
On the problem with subtitles that overlap see [this comment]( https://github.com/iina/iina/issues/4752#issuecomment-1879791704) in issue #4752 which shows how to use the [sub-ass-override](https://mpv.io/manual/stable/#options-sub-ass-override) option as a workaround.

On the keys for changing the subtitle size I have entered issue #4824. That requests such key bindings be added to IINA's defaults. It is possible to configure those keys yourself. Follow these instructions:

- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Key Bindings`
- If you have not already configured a custom set of bindings then click on the `Duplicate` button and provide a name for your bindings in the popup
- In the Key/Action table click on the `+` button
- The `Key Mapping` panel appears
- Click on ⇧g
- Then in the `Or enter mpv command` text box enter `add sub-scale +0.1`
- Click on `OK`
- Repeat setting ⇧f to `add sub-scale -0.1`
- Confirm the panel looks like the screenshot below
- **Restart IINA** to activate the new settings

![issue-4457](https://github.com/iina/iina/assets/86170219/db4139c7-9fc8-4fc6-b7b5-a1d0b8831ca7)
